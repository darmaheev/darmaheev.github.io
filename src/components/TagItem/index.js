import React from "react";
import PropTypes from "prop-types";

import {
  Wrapper,
  DeleteItem
} from "./styles";

const TagItem = ({
  children,
  handleDelete
}) => {
  return (
    <Wrapper>
      {children}
      {typeof handleDelete === "function" && <DeleteItem onClick={() => handleDelete(children)}/>}
    </Wrapper>
  );
};

TagItem.defaultProps = {
  children: "",
  handleDelete: undefined
};

TagItem.propTypes = {
  children: PropTypes.string,
  handleDelete: PropTypes.func
};

export default TagItem;