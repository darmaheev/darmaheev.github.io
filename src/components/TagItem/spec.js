import React from "react";
import TagItem from "./index";
import { MemoryRouter } from "react-router-dom";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders tagItem with default props", () => {
  const tagItem = shallow(<TagItem />);
  expect(tagItem).toMatchSnapshot();
});

it("renders tagItem with data", () => {
  const tagItem = shallow(<TagItem>Tag</TagItem>);
  expect(tagItem.text()).toContain("Tag");
});

it("check tagItem on click", () => {
  const handleDelete = jest.fn();

  const tagItem = mount(
    <MemoryRouter>
      <TagItem handleDelete={handleDelete}>Tag</TagItem>;
    </MemoryRouter>
  );
  const deleteControl = tagItem.find("button").at(0);
  deleteControl.simulate("click");
  expect(handleDelete).toHaveBeenCalledWith("Tag");
});
