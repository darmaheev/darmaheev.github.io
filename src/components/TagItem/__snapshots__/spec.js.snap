// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tagItem with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagItem>
    
  </TagItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "",
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      false,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 27px;
  border-radius: 13px;
  background-color: rgba(31, 151, 201, 0.1);
  font-size: 12px;
  color: #1f97c9;
  font-weight: 400;
  padding: 8px 18px;
  margin-bottom: 8px;
  margin-right: 8px;
  text-transform: uppercase;  
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "",
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        false,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 27px;
  border-radius: 13px;
  background-color: rgba(31, 151, 201, 0.1);
  font-size: 12px;
  color: #1f97c9;
  font-weight: 400;
  padding: 8px 18px;
  margin-bottom: 8px;
  margin-right: 8px;
  text-transform: uppercase;  
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
